pkg/cover/server.go

```diff
@@ -123,24 +125,53 @@ func (s *server) registerService(c *gin.Context) {
 		c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
 		return
 	}
-
+	if service.Network == "" {
+		service.Network = s.NetworkType
+	}
 	u, err := url.Parse(service.Address)
 	if err != nil {
 		c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
 		return
 	}
-	host, port, err := net.SplitHostPort(u.Host)
-	if err != nil {
-		c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
+	if u.Scheme != "https" && u.Scheme != "http" {
+		c.JSON(http.StatusBadRequest, gin.H{"error": "unsupport schema"})
+		return
+	} else if u.Path != "" {
+		c.JSON(http.StatusBadRequest, gin.H{"error": "uri path must empty"})
 		return
 	}
+	if service.Network == "direct" {
+		if u.Host == "" {
+			c.JSON(http.StatusBadRequest, gin.H{"error": "address is empty"})
+			return
+		}
+		if u.Hostname() == "" {
+			c.JSON(http.StatusBadRequest, gin.H{"error": "empty host name"})
+			return
+		}
+	} else {
+		host := u.Hostname()
+		port := u.Port()
+		if host == "" {
+			host = c.ClientIP()
+		}
+		if port == "" {
+			port = "80"
+		}
+		u.Host = fmt.Sprintf("%s:%s", host, port)
+		host, port, err := net.SplitHostPort(u.Host)
+		if err != nil {
+			c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
+			return
+		}
 
-	realIP := c.ClientIP()
-	// only for IPV4
-	// refer: https://github.com/qiniu/goc/issues/177
-	if net.ParseIP(realIP).To4() != nil && host != realIP {
-		log.Printf("the registered host %s of service %s is different with the real one %s, here we choose the real one", service.Name, host, realIP)
-		service.Address = fmt.Sprintf("http://%s:%s", realIP, port)
+		realIP := c.ClientIP()
```

Comment:
I am thinking that no need to change the `register api`, we can just use `goc sever` to control this behave. like, Launch go server with `goc server --ip_revise=false`
And change here's logic as
```go 
	// only do ip revise as needed.
	if s.IPRevise {
		realIP := c.ClientIP()
		// only for IPV4
		// refer: https://github.com/qiniu/goc/issues/177
		if net.ParseIP(realIP).To4() != nil && host != realIP {
			log.Printf("the registered host %s of service %s is different with the real one %s, here we choose the real one", service.Name, host, realIP)
			service.Address = fmt.Sprintf("http://%s:%s", realIP, port)
		}
	}
```